packages/block-library/src/template-part/edit/entity-content/patterns-setup.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * WordPress dependencies
+ */
+import { useDispatch } from '@wordpress/data';
+import {
+	store as blockEditorStore,
+	__experimentalBlockPatternSetup as BlockPatternSetup,
+} from '@wordpress/block-editor';
+import { cloneBlock } from '@wordpress/blocks';
+import { useEffect } from '@wordpress/element';
+
+export default function PatternsSetup( {
+	blockNameWithArea,
+	clientId,
+	setStartBlank,
+} ) {
+	const { replaceInnerBlocks } = useDispatch( blockEditorStore );
+
+	const onSelect = ( blocks ) => {
+		const clonedBlocks = blocks.map( ( block ) => cloneBlock( block ) );
+		replaceInnerBlocks( clientId, clonedBlocks );
+	};
+
+	return (
+		<BlockPatternSetup
+			blockName={ 'core/template-part' }
```

Comment:
Since we are using a custom pattern filtering callback we can omit this.

Reply:
do you think we can omit clientId as well?  I guess its still assumed to be present in the selector so I guess not... but it feels sort of pointless in this circumstance since we are just getting patterns registered to `core/template-part/${variation}` anyways.

Reply:
>do you think we can omit clientId as well?

No because it checks for allowed blocks in the rootClientId that is fetched from the block's clientId